esp/services/ws_fs/ws_fsBinding.cpp

```diff
@@ -370,6 +300,54 @@ IPropertyTree* CFileSpraySoapBindingEx::createPTreeForXslt(const char* method, c
     return pRoot.getClear();
 }
 
+//For every ECLWatchVisible dropzones, read: dropZoneName, directory, and, if available, computer.
+//For every Servers/Server in ECLWatchVisible dropzones, read: directory,
+// server name, and hostname(or IP). Create dropzone("@name", "@directory", "@computer",
+// "@netAddress", "@linux", "@sourceNode") tree into pSoftware.
+void CFileSpraySoapBindingEx::appendDropZones(double clientVersion, IConstEnvironment* env, const char* dfuwuidSourcePartIP, IPropertyTree* softwareTree)
+{
+    IArrayOf<IConstTpDropZone> list;
+    CTpWrapper dummy;
+    dummy.getTpDropZones(clientVersion, nullptr, true, list);
```

Comment:
The CTpWrapper was designed to wrap all of the ESP functions connecting to environment. The getTpDropZones() is added to return a list of TpDropZone objects which is required by WsTopology. The getTpDropZones() also handles legacy dropzone format in environment.xml. Other ESP services (FileSpray for now) use it because I do not have to implement similar logic again.